pkg/kv/kvserver/node_liveness.go

```diff
@@ -1002,6 +1004,39 @@ func (nl *NodeLiveness) GetLivenesses() []kvserverpb.Liveness {
 	return livenesses
 }
 
+// GetLivenessesFromKV returns a slice containing the liveness status of every
+// node on the cluster. It's the improved version of GetLivenesses above, which
```

Comment:
it's not "improved", it serves a different purpose. If being cheap is what you're after, obviously you'd avoid the KV version. We are adding this because we need it for correctness, so a comment could say

> GetLivenessesFromKV returns a slice containing the liveness record of all nodes that have ever been a part of the cluster. The records are read from the KV layer in a KV transaction.

We also need to consider whether this needs to paginate and possibly avoid old node IDs. I hope it does not (i.e. back of the envelope calculations should show that we'll never crack a couple of mbs here).

Reply:
yup, we shouldn't need to paginate here. Updated this comment in #56243.